### PR TITLE
Update to new Portfolio Report API

### PR DESCRIPTION
**Pull request by buchen**
Portfolio Report now

* includes cryptocurrencies
* allows to retrieve all available crypto currencies
* can provides quotes in different currencies

This pull request changes
* remove the now obsolete API to retrieve prices by market
* removes the obsolete properties from the client
* uses Portfolio Report when creating crypto currencies (either via search, the explicit dialog or during the PDF import)

### CODE REVIEW

**Comment by buchen** [comment]
@Nirus2000 please have a look at c1c763147e34c58ab5996d71e693883ec19fcb76 as I needed to change the tests slightly

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java [hunk 1]
```diff
@@ -67,27 +67,29 @@
...
                 {
                     return List.of( //
-                                    new Coin("bitcoin", "BTC", "Bitcoin"), //
-                                    new Coin("ethereum", "ETH", "Ethereum"));
+                                    new Coin("bitcoin", "BTC", "Bitcoin").asResultItem(), //
+                                    new Coin("ethereum", "ETH", "Ethereum").asResultItem());
                 }
-            };
+            });
         }
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java:93]: Can we look into making this a global list of coins tested so far?
Maybe outsource the list?

**Review by Nirus2000 — APPROVED** [review]
I think looks good 👍🏻

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java (hunk 1 above)
Comment (buchen) [reply name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java]: good Idea. I have extracted into a separate helper class with a static method.

Unfortunately, it is a method of the AbstractPDFExtractor which means we still have to override the method in each test.

**Pull request merged**